Packs/SophosXGFirewall/Integrations/sophos_firewall.py

```diff
@@ -0,0 +1,2027 @@
+
+    Args:
+        response (dict): the object to check the error on
+
+    Raises:
+        Exception: if there if an error in the response
+        Exception: if there are no records on list or get
+    """
+    response_message = retrieve_dict_item_recursively(response, '#text')
+    response_code = retrieve_dict_item_recursively(response, '@code')
+    response_status = retrieve_dict_item_recursively(response, 'Status')
+
+    if response_message and 'successful' not in response_message:
+        if response_code and int(response_code) > 299:
+            raise Exception(f'{response_message} (error code: {response_code})')
+    if response_status and 'No. of records Zero.' in response_status:
+        raise Exception(response_status)
+
+
+def generic_add(client: Client, constants: dict, params: dict, builder: Callable) -> CommandResults:
+    """Generic add function, sends false as to_update to generic save function
+
+    Args:
+        client (Client): Sophos XG Firewall Client
+        constants (dict): CONSTANT with the endpoint tag and table headers
+        params (dict): params for the builder
+        builder (Callable): the builder to build the object
+
+    Returns:
+        CommandResults: Command results object
+    """
+    to_update = False
+    return generic_save(client, params, builder, to_update, **constants)
+
+
+def generic_update(client: Client, constants: dict, params: dict,
+                   builder: Callable) -> CommandResults:
+    """Generic update function, sends true as to_update to generic save function
+
+    Args:
+        client (Client): Sophos XG Firewall Client
+        constants (dict): CONSTANT with the endpoint tag and table headers
+        params (dict): params for the builder
+        builder (Callable): the builder to build the object
+
+    Returns:
+        CommandResults: Command results object
+    """
+    to_update = True
+    return generic_save(client, params, builder, to_update, **constants)
+
+
+def generic_save(client: Client, params: dict, builder: Callable, to_update: bool,
+                 endpoint_tag: str, table_headers: list) -> CommandResults:
+    """Generic function for add/update
+
+    Args:
+        client (Client): Sophos XG Firewall Client
+        params (dict): params for the builder
+        builder (Callable): the builder to build the object
+        to_update (bool): True if the object should be updated
+        endpoint_tag (str): The endpoint_tag of the object
+        table_headers (list): table_headers for readable outputs
+
+    Returns:
+        CommandResults: Command results object
+    """
+    data = builder(client, to_update, endpoint_tag, **params)
+    operation = 'update' if to_update else 'add'
+    response = client.set_request((endpoint_tag, data), operation)
+    response = json.loads(xml2json(response.text))
+
+    check_error_on_response(response)
+    return generic_get(client, params.get('name'), endpoint_tag, table_headers)
```

Comment:
Here, I would only update the object and not return a response and call the `generic_get` from the prev func. e.g from `sophos_firewall_app_policy_add_command`.